### PR TITLE
Remove default indent from .editorconfig

### PR DESCRIPTION
**Pull request by pugnascotia**
Prior to 3a3e5f6, there was no default indent configured in `.editorconfig`. This changed to 4 spaces when we configured an explicit indent of 2 for Gradle files. However, this change meant that YAML files then had 4-space indents, which is valid, but the repo's YAML files typically use 2-space indents.

Remove the default indent again, and specify the indent explicitly for a number of file extensions.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -7,10 +7,7 @@ charset = utf-8
 trim_trailing_whitespace = true
 insert_final_newline = true
 indent_style = space
-indent_size = 4
-
-[*.bat]
 indent_size = 2
```
Comment (henningandersen) [inline .editorconfig]: This means xml and html also use 2 space indent. I do not have the full overview of file types, but looks like intellij primarily uses 2 space indent for json, yaml and properties and thus 4 space for the rest. However, asciidoc and sql also seems to be using 2 space indent?

I am not really sure how to proceed without carefully considering each file type that we have. For now, I suggest to specify all the known files explicitly and leave out the default size (though I see the point in eventually including that). Could also keep default indent at 4 (but I think a default of 2 is more appropriate in the end?).

**Review by henningandersen — COMMENTED** [review]
Thanks for looking into this @pugnascotia , I left a comment to consider/discuss.

File: .editorconfig (hunk 1 above)
Comment (henningandersen) [reply .editorconfig]: Reflecting a bit more on this, I think just including xml and html with indent_size 4 into this PR can work out. We should then be prepared to adjust any additional file types too, but hopefully  we have then hit most types correctly.

Comment (pugnascotia) [reply .editorconfig]: I think you're right, I've pushed another change.

**Review by henningandersen — APPROVED** [review]
LGTM.

**Pull request merged**